src/amo/reducers/home.js

```diff
@@ -197,10 +202,33 @@ export const loadHomeData = ({
   };
 };
 
+type LoadMobileHomeDataParams = {|
+  heroShelves: ExternalHeroShelvesType,
+  shelves: { [shelfName: string]: ApiAddonsResponse },
+|};
+
+type LoadMobileHomeDataAction = {|
+  type: typeof LOAD_MOBILE_HOME_DATA,
+  payload: LoadMobileHomeDataParams,
+|};
+
+export const loadMobileHomeData = ({
+  heroShelves,
+  shelves,
+}: LoadMobileHomeDataParams): LoadMobileHomeDataAction => {
+  invariant(shelves, 'shelves is required');
```

Comment:
Why do we need this and why, like the other load function, do we _not_ use `invariant` on `heroShelves`?

Reply:
I think when I was first writing this I didn't realize we needed `heroShelves` for the mobile home page and so I created a separate loader to not have to have conditional logic which controlled when `heroShelves` was mandatory and when it was not. This is probably also the reason there's no `invariant` for it. Looking at it all again now, and seeing that we do need `heroShelves`, I do think it would be simpler to just reuse `loadHomeData`, so I'll try doing that instead.